### PR TITLE
Add the Machine Memory Card (copy paste gadget)

### LINKED ISSUE #281: Allow copying settings between machines
### Cross-mod Integration

_No response_

### Feature Description

Add a feature that allows copying settings between machines.  
Especially useful in conjunction with #280 (slot locking).

Suggested by @xldenis on our discord.

### PR DESCRIPTION
**Pull request by omergunr100**
## What
Lets the item copy and paste single-block machine configuration, including item/fluid output direction (relative to front), whether auto output is enabled, whether you can input from output side, muffle state.

Also adds the base functionality tooltip to the terminal about automatically building multi-blocks.

## Implementation Details
Just another interaction item behaviour.

## Note
If accepted, this pr will close #1476 and resolves #281

### CODE REVIEW

**Comment by omergunr100** [comment]
todo:
Ghost suggested making it the functionality of a new item and is making a texture for it.

Do not merge until he does his thing.

File: src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+            tooltipComponents.add(Component.literal(""));
+            var tag = stack.getOrCreateTag();
+            if (tag.contains(CONFIG_DATA)) {
+                var data = tag.getCompound(CONFIG_DATA);
+                if (data.contains(ORIGINAL_FRONT)) {
+                    var origFront = intToDirection(data.getInt(ORIGINAL_FRONT));
+                    if (data.contains(ITEM_CONFIG)) {
+                        var itemData = data.getCompound(ITEM_CONFIG);
+                        tooltipComponents.add(Component.translatable("behaviour.setting.output.direction.tooltip",
+                                "§6Item§r",
```
Comment (YoungOnionMC) [inline src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java]: please change all these literal texts into lang for translation purposes

File: src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+                        tooltipComponents.add(Component.translatable("behaviour.setting.item_auto_output.tooltip",
+                                "§6Item§r", (itemData.getBoolean(AUTO) ? "§aenabled§r" : "§4disabled§r")));
+                        tooltipComponents.add(Component.translatable(
+                                "behaviour.setting.allow.input.from.output.tooltip", "§6Item§r",
+                                (itemData.getBoolean(INPUT_FROM_OUTPUT_SIDE) ? "§aenabled§r" : "§4disabled§r")));
+                    }
+                    if (data.contains(FLUID_CONFIG)) {
+                        var fluidData = data.getCompound(FLUID_CONFIG);
+                        tooltipComponents.add(Component.translatable("behaviour.setting.output.direction.tooltip",
+                                "§9Fluid§r", "§e" + relativeDirectionString(origFront,
```
Comment (YoungOnionMC) [inline src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java]: change relativeDirectionString() to also return a translatable

**Review by YoungOnionMC — CHANGES_REQUESTED** [review]

File: src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java (hunk 1 above)
Comment (omergunr100) [reply src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java]: done

File: src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java (hunk 2 above)
Comment (omergunr100) [reply src/main/java/com/gregtechceu/gtceu/common/item/tool/behavior/MetaMachineConfigCopyBehaviour.java]: done

**Review by YoungOnionMC — APPROVED** [review]

**Pull request merged**